### ISSUE TITLE
BigQuery: Newline character gets inserted multiple times on subsequent runs

### ISSUE THREAD

**Issue by tuzonghua** [sqlfluff/sqlfluff#5542]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running sqlfluff on a series of relatively simple BigQuery files resulted in a bug where sqlfluff prompted me to fix the files multiple times, each time inserting a newline.

### Expected Behaviour

The sample file gets fixed only once and not multiple times.

### Observed Behaviour

The sample SQL code gets fixed multiple times, each pass adding another `\n`.

First pass:
```
$ sqlfluff fix test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:   2 | P:   1 | LT05 | Line is too long (139 > 120).
                       | [layout.long_lines]
==== fixing violations ====
1 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
WARNING    Skipping edit patch on uncertain templated section [('mid_point', slice(14, 41, None))], Please report this warning on GitHub along with the query that produced it.
== [test.sql] FIXED
Done. Please check your files to confirm.
All Finished 📜 🎉!
```

First pass output SQL:
```sql
SELECT *
FROM 
    `{{ params.some_project }}.{{ params.some_long_dataset_name }}.{{ params.some_long_name_table_id }}`
WHERE some_col = {{ params.some_val }}
```

Second pass:
```
$ sqlfluff fix test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:   2 | P:   5 | LT01 | Unnecessary trailing whitespace.
                       | [layout.spacing]
L:   3 | P:   1 | LT02 | Line should not be indented.
                       | [layout.indent]
L:   3 | P:  32 | LT05 | Line is too long (134 > 120).
                       | [layout.long_lines]
==== fixing violations ====
3 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
WARNING    Skipping edit patch on uncertain templated section [('mid_point', slice(15, 46, None))], Please report this warning on GitHub along with the query that produced it.
== [test.sql] FIXED
Done. Please check your files to confirm.
All Finished 📜 🎉!
```

Second pass output SQL:
```sql
SELECT *
FROM
    
    `{{ params.some_project }}.{{ params.some_long_dataset_name }}.{{ params.some_long_name_table_id }}`
WHERE some_col = {{ params.some_val }}
```

Third pass:
```
$ sqlfluff fix test.sql
==== finding fixable violations ====
== [test.sql] FAIL
L:   3 | P:   1 | LT01 | Unnecessary trailing whitespace.
                       | [layout.spacing]
L:   4 | P:   1 | LT02 | Line should not be indented.
                       | [layout.indent]
L:   4 | P:  32 | LT05 | Line is too long (134 > 120).
                       | [layout.long_lines]
==== fixing violations ====
3 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
WARNING    Skipping edit patch on uncertain templated section [('mid_point', slice(19, 50, None))], Please report this warning on GitHub along with the query that produced it.
== [test.sql] FIXED
Done. Please check your files to confirm.
All Finished 📜 🎉!
```

Third pass output SQL:
```sql
SELECT *
FROM

    
    `{{ params.some_project }}.{{ params.some_long_dataset_name }}.{{ params.some_long_name_table_id }}`
WHERE some_col = {{ params.some_val }}
```

### How to reproduce

Test SQL:
```sql
SELECT *
FROM `{{ params.some_project }}.{{ params.some_long_dataset_name }}.{{ params.some_long_name_table_id }}`
WHERE some_col = {{ params.some_val }}
```

Test config:
```
[sqlfluff]
dialect = bigquery
templater = jinja
max_line_length = 120

[sqlfluff:templater:jinja:context]
params = {"some_project": "some_name", "some_long_dataset_name ": "some_other_name", "some_long_name_table_id": "yet_another_name", "some_val": "some_val"}
```

Steps to reproduce:

1. Run `sqlfluff fix test.sql` using the test config 
2. Run the command multiple times, observing the extra blank lines being inserted with each run

### Dialect

BigQuery

### Version

```
sqlfluff, version 2.3.5
Python 3.8.18
```

### Configuration

```
[sqlfluff]
dialect = bigquery
templater = jinja
max_line_length = 120

[sqlfluff:templater:jinja:context]
params = {"some_project": "some_name", "some_long_dataset_name ": "some_other_name", "some_long_name_table_id": "yet_another_name", "some_val": "some_val"}
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tuzonghua** [comment]
Doesn't occur in 3.0.5
